elastalert/test_rule.py

```diff
@@ -222,10 +217,9 @@ def get_id():
             endtime = ts_now()
             starttime = endtime - datetime.timedelta(days=args.days)
 
-        # Set run_every to cover the entire time range unless count query, terms query or agg query used
-        # This is to prevent query segmenting which unnecessarily slows down tests
-        if not rule.get('use_terms_query') and not rule.get('use_count_query') and not rule.get('aggregation_query_element'):
-            conf['run_every'] = endtime - starttime
+        # Set run_every to cover the entire time range, this is to prevent query segmenting which unnecessarily slows down tests
+        # (see elastalert.get_segment_size())
+        conf['run_every'] = endtime - starttime
```

Comment:
For the case where you have multiple rules it's not meanigful to say if you're using terms / count / agg (some rules might be, other rules might not). 

From a rough eyeball of the code, the `run_rule()` method that we use to test with only looks at `run_every` via `set_starttime()` and `get_segment_size()`. `set_starttime()` only uses `run_every` if the previous endtime is set, which it wouldn't be in this test (I think). `get_segment_size()` only applies to terms / count / agg queries anyway.

So basically I concluding that setting this parameter wouldn't do any harm.